packages/shorebird_cli/lib/src/commands/release_new/release_pipeline.dart

```diff
@@ -0,0 +1,347 @@
+      appId: appId,
+      releaseVersion: version,
+    );
+
+    if (existingRelease != null) {
+      codePushClientWrapper.ensureReleaseIsNotActive(
+        release: existingRelease,
+        platform: releaseType.releasePlatform,
+      );
+
+      // All artifacts associated with a given release must be built
+      // with the same Flutter revision.
+      final errorMessage = '''
+${styleBold.wrap(lightRed.wrap('A release with version $version already exists but was built using a different Flutter revision.'))}
+
+  Existing release built with: ${lightCyan.wrap(existingRelease.flutterRevision)}
+  Current release built with: ${lightCyan.wrap(flutterVersion)}
+
+${styleBold.wrap(lightRed.wrap('All platforms for a given release must be built using the same Flutter revision.'))}
+
+To resolve this issue, you can:
+  * Re-run the release command with "${lightCyan.wrap('--flutter-version=${existingRelease.flutterRevision}')}".
+  * Delete the existing release and re-run the release command with the desired Flutter version.
+  * Bump the release version and re-run the release command with the desired Flutter version.''';
+      throw BuildPipelineException(
+        message: errorMessage,
+        exitCode: ExitCode.software,
+      );
+    }
+  }
+
+  /// Prints a confirmation prompt with details about the release to be created.
+  /// If the user confirms, the release will be created. If the user cancels,
+  /// the command will exit with a success code. When running in a headless
+  /// or CI environment, this prompt will print but will not wait for user
+  /// confirmation.
+  Future<void> confirmCreateRelease({
+    required AppMetadata app,
+    required String releaseVersion,
+    required String flutterVersion,
+  }) async {
+    final flutterVersionString = await shorebirdFlutter.getVersionAndRevision();
+    // TODO(bryanoltman): include archs in the summary for android
+    // (and other platforms?)
+    final summary = [
+      '''📱 App: ${lightCyan.wrap(app.displayName)} ${lightCyan.wrap('(${app.appId})')}''',
+      if (flavor != null) '🍧 Flavor: ${lightCyan.wrap(flavor)}',
+      '📦 Release Version: ${lightCyan.wrap(releaseVersion)}',
+      '🕹️  Platform: ${lightCyan.wrap(releaseType.releasePlatform.name)}',
+      '🐦 Flutter Version: ${lightCyan.wrap(flutterVersionString)}',
+    ];
+
+    logger.info('''
+
+${styleBold.wrap(lightGreen.wrap('🚀 Ready to create a new release!'))}
+
+${summary.join('\n')}
+''');
+
+    if (shorebirdEnv.canAcceptUserInput) {
+      final confirm = logger.confirm('Would you like to continue?');
+
+      if (!confirm) {
+        throw BuildPipelineException(
+          message: 'Aborting.',
+          exitCode: ExitCode.success,
+        );
+      }
+    }
+  }
+
+  /// Fetches the release with version [version] from the server or creates a
+  /// new release if none exists.
+  Future<Release> getOrCreateRelease({required String version}) async {
```

Comment:
The async/awaits here are not needed, as you can just return the returned future from the calls

Reply:
In this case they are, because we need to await `maybeGetRelease` to know whether we need to call `createRelease`